src/unix/fs.c

```diff
@@ -1234,11 +1234,11 @@ static int uv__fs_statx(int fd,
 
   rc = uv__statx(dirfd, path, flags, mode, &statxbuf);
 
-  if (rc == -1) {
+  if (rc != 0) {
```

Comment:
This change is unsound, the value of errno is only defined when `rc == -1`.

Reply:
I can revert it back back to my original change where I had an `else` statement checking `else if (rc == 1) {`. Above change has been made based on saghul's review.

Having said that, `rc` values in this sys call are supposed to always either be `0` or `-1`, so any other value other than `0` can be considered as an error.

Reply:
Yes, but system call wrappers only set errno when the return value is -1.

A comment explaining under what conditions `rc == 1` has been observed would be helpful too because it just looks wrong without context.

Aside: I wonder if the system call number is correct since it returns _something_ rather than `ENOSYS`. Libuv currently calls system call 379 on s390x.

Reply:
System call seems to be correct, taken from zLinux shows this unimplemented comment:
```
/usr/include/asm/unistd.h:308:/* Number 379 is reserved for sys_statx */
```

I have added a comment explaining the change in more detail.